src/grafana_agent.py

```diff
@@ -880,6 +883,14 @@ def _tracing_receivers(self) -> Dict[str, Union[Any, List[Any]]]:
         Returns:
             a dict with the receivers config.
         """
+        if not self._tracing.is_ready():
+            # we need to guard against the "upstream" tracing backend relation as grafana-agent fails to start
+            # if we don't provide the tracing endpoint to send data to.
```

Comment:
Emm, I think the comment is correct, but is in the wrong function maybe?
Here in _tracing_receivers, what we dont want to generate receivers config if we can generate the exporter config

If you get the feeling both are essentially saying the same thing, feel free to ignore and blame it on a late night review :D